cms/models/placeholdermodel.py

```diff
@@ -299,4 +306,81 @@ def actions(self):
             self._actions_cache = getattr(field, 'actions', PlaceholderNoAction())
         return self._actions_cache
 
+    def get_cache_expiration(self, request, response_timestamp):
+        """
+        Returns the number of seconds (from «response_timestamp») that this
+        placeholder can be cached. This is derived from the plugins it contains.
+
+        This method must return: EXPIRE_NOW <= int <= MAX_EXPIRATION_IN_SECONDS
+
+        :type request: HTTPRequest
+        :type response_timestamp: datetime
+        :rtype: int
+        """
+        lang = getattr(request, 'lang', None)
+
+        min_ttl = MAX_EXPIRATION_TTL
+        for plugin_item in self.get_plugins(lang):
+            instance, plugin = plugin_item.get_plugin_instance()
```

Comment:
@mkoistinen I'm a bit curious to know how you measured the performance here.
I've used the django debug toolbar's `sqldebugshell` to isolate the performance for this specific method on the Placeholder model.

Here's an example with the current line:

https://gist.github.com/czpython/32d639e3cd17da64f7ba

And this one is with that line refactored to match my suggestion:

https://gist.github.com/czpython/cd3b52e298e424b15de2